File: python/2.7/observer/CurrentConditionsDisplay.py
from Observer import *
from DisplayElement import *

class CurrentConditionsDisplay(Observer, DisplayElement):
    def __init__(self, weatherData):
        self.temperature = 0.0
        self.humidity = 0.0
        self.weatherData = weatherData
        self.weatherData.registerObserver(self)

    def update(self, temperature, humidity, pressure):
        self.temperature = temperature
        self.humidity = humidity
        self.display()

    def display(self):
        print "Current conditions: %s F degrees and %s humidity" %(self.temperature, self.humidity)

File: js/node/observer/CurrentConditionsDisplay.js
require('./Observer');
require('./DisplayElement');

CurrentConditionsDisplay = function(weatherData) {
    Observer.apply(this);
    DisplayElement.apply(this);
    this.temperature = 0.0;
    this.humidity = 0.0;
    this.weatherData = weatherData;
    this.weatherData.registerObserver(this);
    this.update = function(temperature, humidity, pressure) {
        this.temperature = temperature;
        this.humidity = humidity;
        this.display();
    }
    this.display = function() {
        console.log("Current conditions: " + this.temperature + " F degrees and " + this.humidity + " humidity");
    }
}